File: librdp/src/main/java/com/freerdp/freerdpcore/application/RdpScreenReceiver.java
package com.freerdp.freerdpcore.application;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class RdpScreenReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.getAction().equals(Intent.ACTION_SCREEN_OFF)) {
            RdpApp.startDisconnectTimer();
        } else if (intent.getAction().equals(Intent.ACTION_SCREEN_ON)) {
            RdpApp.cancelDisconnectTimer();
        }
    }
}

File: librdp/src/main/java/com/freerdp/freerdpcore/application/RdpScreenReceiver.kt
package com.freerdp.freerdpcore.application

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

/**
 * 亮屏息屏
 *
 * @author why
 */
class RdpScreenReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        when (intent.action) {
            Intent.ACTION_SCREEN_OFF -> {
                RdpApp.startDisconnectTimer()
            }
            Intent.ACTION_SCREEN_ON -> {
                RdpApp.cancelDisconnectTimer()
            }
        }
    }
}